Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -27,6 +38,94 @@ void RegExpConstructor::initialize(Realm& realm)
+    // dbgln("startIndex = {}, endIndex = {}, len = {}", startIndex, endIndex, len);
+
+    // 4. Assert: startIndex and endIndex are integers such that 0 ≤ startIndex ≤ endIndex ≤ len.
+    // VERIFY(0 <= startIndex);
+    VERIFY(startIndex <= endIndex);
+    VERIFY(endIndex <= len);
+
+    // 5. Assert: capturedValues is a List of Strings.
+
+    // 6. Let n be the number of elements in capturedValues.
+    size_t n = capturedValues.size();
+
+    // 7. Set the value of C’s [[RegExpInput]] internal slot to S.
+    // dbgln("input = {}, n = {}", S.to_utf8(), n);
+    m_legacy_properties.set_input(S);
+
+    // 8. Set the value of C’s [[RegExpLastMatch]] internal slot to a String whose length is endIndex - startIndex and containing the code units from S with indices startIndex through endIndex - 1, in ascending order.
+    auto const& lastMatch = Utf16String(S.view().substring_view(startIndex, endIndex - startIndex));
+    // dbgln("lastMatch = {}", lastMatch.to_utf8());
+    m_legacy_properties.set_last_match(lastMatch);
+
+    // 9. If n > 0, set the value of C’s [[RegExpLastParen]] internal slot to the last element of capturedValues.
+    if (n > 0) {
+        auto& item = capturedValues[n - 1];
+        // dbgln("lastParen = {}", item.to_utf8());
+        m_legacy_properties.set_last_paren(Utf16String(item));
+    }
+    // 10. Else, set the value of C’s [[RegExpLastParen]] internal slot to the empty String.
+    else {
+        auto const& lastParen = Utf16String(String::empty().view());
+        // dbgln("lastParen = {}", lastParen.to_utf8());
+        m_legacy_properties.set_last_paren(lastParen);
+    }
+
+    // 11. Set the value of C’s [[RegExpLeftContext]] internal slot to a String whose length is startIndex and containing the code units from S with indices 0 through startIndex - 1, in ascending order.
+    auto const& leftContext = Utf16String(S.view().substring_view(0, startIndex));
+    // dbgln("leftContext = {}", leftContext.to_utf8());
+    m_legacy_properties.set_left_context(leftContext);
+
+    // 12. Set the value of C’s [[RegExpRightContext]] internal slot to a String whose length is len - endIndex and containing the code units from S with indices endIndex through len - 1, in ascending order.
+    auto const& rightContext = Utf16String(S.view().substring_view(endIndex, len - endIndex));
+    // dbgln("rightContext = {}", rightContext.to_utf8());
+    m_legacy_properties.set_right_context(rightContext);
+
+    // 13. For each integer i such that 1 ≤ i ≤ 9
+    for (size_t i = 0; i < n; i++) {
+        // If i ≤ n, set the value of C’s [[RegExpPareni]] internal slot to the ith element of capturedValues.
+        auto& value = capturedValues[i];
+        // dbgln("${} = {}", (i + 1), value.to_utf8());
+        m_legacy_properties.set_paren(i + 1, Utf16String(value));
+
+        // Else, set the value of C’s [[RegExpPareni]] internal slot to the empty String.
+        // Ignore the else branch
```

Comment:
If we don't do what the spec says use
`// NOTE: We don't need to perform the else branch because....`